components/proxy/src/main/java/com/hotels/styx/routing/handlers/StandardHttpPipeline.java

```diff
@@ -77,19 +77,19 @@ public StandardHttpPipeline(List<HttpInterceptor> interceptors, HttpHandler hand
         }
 
         @Override
-        public StyxObservable<HttpResponse> proceed(HttpRequest request) {
+        public Eventual<HttpResponse> proceed(HttpRequest request) {
             if (index < interceptors.size()) {
                 HttpInterceptor.Chain chain = new HttpInterceptorChain(this, index + 1);
                 HttpInterceptor interceptor = interceptors.get(index);
 
                 try {
                     return interceptor.intercept(request, chain);
                 } catch (Throwable e) {
-                    return StyxObservable.error(e);
+                    return Eventual.error(e);
                 }
             }
-            return fromRxObservable(toRxObservable(client.handle(request, this.context))
-                    .compose(StandardHttpPipeline::sendErrorOnDoubleSubscription));
+            return new Eventual<>(toPublisher(RxReactiveStreams.toObservable(client.handle(request, this.context))
```

Comment:
Am I reading this correctly? You are transforming from publisher to observable and then back to publisher?

Reply:
It first turns the `Eventual` from `client.handle` into an `rx.Observable`, so that `sendErrorOnDoubleSubscription` can be composed in. After that the composed `rx.Observable` is converted back to publisher.

This needs to be tidied up but that could be a separate enhancement.

Reply:
You could use lines + indentation to show which bits happen at which stage.

Reply:
It could be written line this, to show which parts happen when:
```
return new Eventual<>(
                    toPublisher(
                            toObservable(client.handle(request, this.context))
                                    .compose(StandardHttpPipeline::sendErrorOnDoubleSubscription)
                    )
            );
```